tools/cli/util.go

```diff
@@ -904,25 +905,67 @@ func showNextPage() bool {
 // paginate creates an interactive CLI mode to control the printing of items
 func paginate(c *cli.Context, paginationFn collection.PaginationFn) error {
 	more := c.Bool(FlagMore)
+	isTableView := !c.Bool(FlagPrintJSON)
 	pageSize := c.Int(FlagPageSize)
 	if pageSize == 0 {
 		pageSize = defaultPageSize
 	}
 	iter := collection.NewPagingIterator(paginationFn)
 
-	pageItemsCount := 0
+	var pageItems []interface{}
 	for iter.HasNext() {
-		batch, err := iter.Next()
+		item, err := iter.Next()
 		if err != nil {
 			return err
 		}
 
-		prettyPrintJSONObject(batch)
-		pageItemsCount++
-		if pageItemsCount%pageSize == 0 && (!more || !showNextPage()) {
-			break
+		pageItems = append(pageItems, item)
+		if len(pageItems) == pageSize {
+			if isTableView {
+				printTable(pageItems)
+			} else {
+				prettyPrintJSONObject(pageItems)
+			}
+
+			if !more || !showNextPage() {
+				break
+			}
+			pageItems = pageItems[:0]
+		}
+	}
+
+	return nil
+}
+
+func printTable(items []interface{}) error {
+	if len(items) == 0 {
+		fmt.Println("no items to print")
+		return nil
+	}
+
+	var fields []string
+	e := reflect.ValueOf(items[0]).Elem()
+	t := e.Type()
+	for i := 0; i < e.NumField(); i++ {
+		fields = append(fields, t.Field(i).Name)
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetBorder(false)
+	table.SetColumnSeparator("|")
+	table.SetHeader(fields)
+	table.SetHeaderLine(false)
+	for i := 0; i < len(items); i++ {
+		item := reflect.ValueOf(items[i]).Elem()
```

Comment:
This will panic if `items[i]` is not a pointer but `string` for example.

Reply:
Another case is `***string`. Do you want to support it?

Reply:
addressed both